packages/remix-express/server.ts

```diff
@@ -94,8 +94,7 @@ export function createRemixRequest(
   let port = hostnamePort || hostPort;
   // Use req.hostname here as it respects the "trust proxy" setting
   let resolvedHost = `${req.hostname}${port ? `:${port}` : ""}`;
-  // TODO: could it break existing hacky custom base use cases?
-  let url = new URL(`${req.protocol}://${resolvedHost}${req.originalUrl}`);
+  let url = new URL(`${req.protocol}://${resolvedHost}${req.url}`);
```

Comment:
I didn't seem to need this in my testing of a remix vite express app using a basename - was this intended for apps using custom middlewares that mess with `req.url`?

Reply:
This probably depends on the way express routers/middlewares are structured.

In my example setup https://github.com/hi-ogawa/remix-vite-custom-base/pull/1, I used a separate `express.Router` to handle remix related things completely separately, which looks like this:

```ts
// [ROOT ROUTER] handle "/..."
const rootApp = express()

// things not related to remix goes here...
rootApp.get("/", ...) // 
rootApp.get("/not-remix", ...) 

// [SUB ROUTER] handle "/mybase/..." as a separate route for remix
const remixApp = express.Router()
rootApp.use("/mybase", remixApp)

// usual remix custom server setup goes here...
remixApp.use(viteDevServer.middlewares)
remixApp.all("*", createRequestHandler({...}));

// finally listen at the root
rootApp.listen()
```

In this case, `url = strip(originalUrl, "/mybase")` is probably done by express when nesting router, so this would be "custom middleware that mess with `req.url`" like you suggested.

I'm trying to find how to avoid such stripping by middlewares, but I cannot figure it out. For example, replacing above `remixApp` with `rootApp` to avoid intermediate router doesn't seem to work, which is probably because of `viteDevServer.middlewares` also strips basename https://github.com/vitest-dev/vite/blob/06494443194763d6dd125740961cec82dd9a29d6/packages/vite/src/node/server/middlewares/base.ts#L21-L26.

Could you share how your express server is setup?

Reply:
ah ok yeah I had just a very simple setup with static base paths and no middlewares/sub-routers: 

```js
const app = express();

app.use("/base/", viteDevServer?.middlewares || express.static("build/client"));

app.all(
  "/base/*",
  createRequestHandler({
    build: viteDevServer
      ? () => viteDevServer.ssrLoadModule("virtual:remix/server-build")
      : await import("./build/server/index.js"),
  })
);
```

I like that workaround you found to re-assign so I agree that it's safest to stick with `req.url` now and if folks are messing with the URL ahead of time they can reset it 👍